src/cache_manager.cc

```diff
@@ -248,6 +248,8 @@ CacheManager::ParseHeaders(const HttpRequest * request, Mgr::ActionParams &param
 
```

Comment:
This PR assumes that future Actions should react to Accept request header values, presumably producing output of two (or more) different Content-Types. Why is that a good idea? Should not all Actions simply respond using YAML instead, disregarding the Accept request header value (or validating it centrally, before the Action is even created and forwarded to workers)? If we have already agreed to proceed with this design, please feel free to point me to that agreement, of course.

Reply:
The ad-hoc format accepted by cachemgr.cgi is not fully compliant YAML. So to retain support for tools expecting that syntax we should only send YAML when it is indicated by the client Accept header.

IIRC, we agreed to the principal of moving to YAML, but still have disagreement on the specifics.

Reply:
> The ad-hoc format accepted by cachemgr.cgi is not fully compliant YAML.

Not sure why you are talking about cachemgr.cgi as if it specifies what output Squid does or should produce. That CGI script is just a hack that we fully control. It will accept whatever we make it accept. It does not and should not determine Squid output format(s). I do agree that Squid output is not YAML today, of course, but that fact has nothing to do with cachemgr.cgi.

> So to retain support for tools expecting that syntax we should only send YAML when it is indicated by the client Accept header.

IMO, we do not want to add code that works towards retaining support for tools requiring non-YAML syntax because we do not want to create and maintain code that generates responses in _two_ different formats, especially when one of those formats is an undefined/unstructured/ad-hoc/custom/informal format. We want to move towards YAML, not towards YAML+informal. We cannot get to YAML immediately, of course, but I see no value (and quite a bit of harm) in enabling this kind of Action code:

```C++
if (clientAcceptsYaml)
    generateYamlResponse();
else
    generateInformalResponse();
```

And if the above is not what we want, then I see no value in telling Actions what formats the client accepts.

> IIRC, we agreed to the principal of moving to YAML

Yes, that matches my recollection as well.

> but still have disagreement on the specifics.

That is probably true, but this change request disagreement appears to be about the very target/direction of that move rather than any move _specifics_. If we target just one format, then Actions do not need to know what formats the client accepts.